### ISSUE TITLE
[expo-av] Sound status shows incorrect duration for MP3 file processed by ffmpeg

### ISSUE THREAD

**Issue by Bobberclobber** [expo/expo#16139]
### Summary

Part of my app involves concatenating MP3 files into one long file. I do this using an AWS Lambda which runs the following ffmpeg command: `ffmpeg -f concat -safe 0 -i INPUT -write_xing 0 -id3v2_version 0 -c copy -f mp3 OUTPUT`. As far as I can tell, the output file is a normal MP3 file, and it behaves as such in the media players I've tested, but when I load the file using expo-av, the status callback contains an incorrect duration causing all sorts of problems during playback. This happens even when using just a single file as input. Since the output files play fine outside of expo-av I'm assuming that the incorrect duration is calculated at some point during the loading process, but during my debugging I have not been able to find exactly when this happens.

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

42

### Environment

  Expo CLI 5.0.3 environment info:
    System:
      OS: Windows 10 10.0.19043
    Binaries:
      Node: 14.17.3 - C:\Program Files\nodejs\node.EXE
      Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
      npm: 6.14.13 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: Version  4.1.0.0 AI-201.8743.12.41.7042882
    npmPackages:
      expo: ~42.0.1 => 42.0.3
      react: 16.13.1 => 16.13.1
      react-dom: 16.13.1 => 16.13.1
      react-native: https://github.com/expo/react-native/archive/sdk-42.0.0.tar.gz => 0.63.2
      react-native-web: ^0.17.1 => 0.17.1
    Expo Workflow: managed

### Reproducible demo

https://snack.expo.dev/@jgfager/calm-blueberries

The demo above shows the mismatched duration when loading an audio file before (test.mp3) and after (test_ffmpeg.mp3) being passed through ffmpeg. If you download the files and listen to them you will find that the content is identical as far as the human ear can tell, and the sizes show only very minor differences which I (maybe incorrrectly) attribute to ffmpeg making some changes to some metadata. I have noticed that the bitrate is different, but it is not enough to explain the several seconds of difference in duration I observe (11 seconds in the original and 7 seconds in the processed file). The duration of the processed file is also not random (it's the same every time), and is never updated during playing if we look in a callback attached to the playback.

**Comment by AdamGerthel** [comment]
This sounds similar, but not identical, to https://github.com/expo/expo/issues/16139

**Comment by mnightingale** [comment]
See https://github.com/google/ExoPlayer/issues/6698

You can also see the problem if you play the file in something like VLC, duration is initially 7s, then 10s and finally 11s.

tl;dr variable bitrate file without Xing header.

Try removing `-write_xing 0 -id3v2_version 0` but then I think duration will be estimated from bitrate, I guess you might need to investigate a better ffmpeg concat method (sorry can't help with that).
